app/views/questions/show.html.erb

```diff
@@ -0,0 +1,68 @@
+<h2><%= @question.question_title %></h2>
+<p><%= @question.content %></p>
+
+<% if logged_in? %>
+  <%= form_for @question, url: question_votes_path(@question), method: :post do |f| %>
+    <%= f.hidden_field :user_id, value: current_user.id %>
+    <%= f.submit "upvote" %>
+  <% end %>
+
+  <%= form_for @question, url: question_votes_path(@question), method: :post do |f| %>
```

Comment:
for both of these, `button_to question_votes_path(@question, upvote="upvote")` would create the form for you.